lib/http/errors.rb

```diff
@@ -20,4 +20,7 @@ class TimeoutError < Error; end
 
   # Header name is invalid
   class InvalidHeaderNameError < Error; end
+
+  # Header value is of unexpected format (similar to Net::HTTPHeaderSyntaxError)
+  class HeaderSyntaxError < Error; end
```

Comment:
Mmm. I tend to agree that single `class HeaderError < Error; end` should be enough.
But in order to make it backward compatible I propose to do this:

``` ruby
class HeaderError < Error; end

# @deprecated
InvalidHeaderNameError = HeaderError
```